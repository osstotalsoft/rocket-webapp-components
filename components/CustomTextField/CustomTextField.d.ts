import { InputAdornmentProps } from "@material-ui/core";
import { StandardTextFieldProps, FilledTextFieldProps, OutlinedTextFieldProps } from "@material-ui/core/TextField";
import React from "react";

export interface Props {
    /**
     * Override or extend the styles applied to the component.
     */
    className?: string
    /**
     * @default false
     * If true, the input will take up the full width of its container.
     */
    fullWidth?: boolean
    /**
     * The variant to use.
     */
    variant?: 'filled' | 'outlined' | 'standard'
    /**
     * If dense or normal, will adjust vertical spacing of this and contained components.
     */
    margin?: 'dense' | 'normal' | 'none'
    /**
     * @default false
     * If true, the label is displayed as required and the input element` will be required.
     */
    required?: boolean
    /**
     * @default false
     * If true, a textarea element will be rendered instead of an input.
     */
    multiline?: boolean
    /**
     * Minimum number of rows to display when multiline option is set to true.
     */
    minRows?: number | string
    /**
     * Maximum number of rows to display when multiline option is set to true.
     */
    maxRows?: number | string
    /**
     * @default false
     * If true, the input will accept only numeric values.
     */
    isNumeric?: boolean
    /**
     * Other properties you can provide to the Input component.
     */
    customInputProps?: Object
    /**
     * Attributes applied to the input element.
     */
    inputProps?: Object
    /**
     * The short hint displayed in the input before the user enters a value.
     */
    placeholder?: string
    /**
     * The adornment of componenent. (Usually an InputAdornment from material-ui)
     */
    endAdornment?: InputAdornmentProps
    /**
     * Start adornment of componenent. (Usually an InputAdornment from material-ui)
     */
    startAdornment: InputAdornmentProps
    /**
     * The label content.
     */
    label?: React.ReactNode
    /**
     * Props applied to the InputLabel element.
     */
    InputLabelProps?: Object
    /**
     * The value of the `input` element, required for a controlled component.
     */
    value?: unknown
    /**
     * The default value of the input element.
     */
    defaultValue?: unknown
    /**
     * @default false
     * If true, the input element will be disabled.
     */
    disabled?: boolean
    /**
     * @default false
     * If true, the label will be displayed in an error state.
     */
    error?: boolean
    /**
     * The helper text content.
     */
    helperText?: React.ReactNode
    /**
     * Callback fired when the value is changed.
     * @default  () => { }
     * @param {object} event The event source of the callback.
     * You can pull out the new value by accessing `event.target.value` (string).
     */
    onChange?: (object: Object) => void
    /**
     * @default 0
     * The delay of debouncing.
     */
    debounceBy?: number
    /**
     * The current language, preferably taken from the i18next (i18.language) or another internationalization library
     */
    language: string
}

export type CustomTextFieldProps = Props | StandardTextFieldProps | FilledTextFieldProps | OutlinedTextFieldProps;

/**
 *
 * Demos:
 *
 * - https://bit.dev/totalsoft_oss/react-mui/custom-text-field
 *
 *  Take a look over the code for further information:https://github.com/osstotalsoft/rocket-webapp-components/blob/master/components/CustomTextField/CustomTextField.js
 * 
 * Material-UI component used:
 * 
 * - TextField: https://material-ui.com/api/text-field/
 * 
 */
export default function CustomTextField(props: CustomTextFieldProps): JSX.Element;