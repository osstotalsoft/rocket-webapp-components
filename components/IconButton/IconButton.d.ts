import { IconButtonProps as BaseProps } from '@material-ui/core/IconButton';

export interface IconButtonProps extends Omit<BaseProps, 'color'> {
    /**
    * The content of the button
    */
    children?: React.ReactNode
    /**
    * @default "theme"
    * The color of the button
    */
    color?: "primary" |
    "info" |
    "theme" |
    "themeNoBackground" |
    "themeWithBackground" |
    "success" |
    "warning" |
    "danger" |
    "rose" |
    "white" |
    "simple" |
    "defaultNoBackground" |
    "primaryNoBackground" |
    "infoNoBackground" |
    "successNoBackground" |
    "warningNoBackground" |
    "dangerNoBackground" |
    "roseNoBackground"
    /**
    * @default "medium"
    * The size of the button
    */
    size?: "small" | "medium"
    /**
    * Override or extend the styles applied to the component
    */
    className?: string
    /**
    * @default false
    * If true, button will be disabled.
    */
    disabled?: boolean
    /**
    * @default false
    * If true, the keyboard focus ripple will be disabled. 
    */
    disableFocusRipple?: boolean
    /**
    * If true, the ripple effect will be disabled.
    */
    disableRipple?: boolean
    /**
    * @default "false"
    * If given, uses a negative margin to counteract the padding on one side 
    * (this is often helpful for aligning the left or right side of the icon 
    * with content above or below, without ruining the border size and shape).
    */
    edge?: "start" | "end" | "false"
    /**
    * The text to be displayed when the user hovers over the button
    */
    tooltip?: string
    /**
    * Demos:
    * 
    * - https://bit.dev/totalsoft_oss/react-mui/icon-button
    * 
    *  Take a look over the code for further information:https://github.com/osstotalsoft/rocket-webapp-components/blob/master/components/IconButton/IconButton.js
    * 
    * Material-UI component used:
    * 
    * - IconButton: https://material-ui.com/api/icon-button/
    * 
    */
}

export default function IconButton(props: IconButtonProps): JSX.Element;