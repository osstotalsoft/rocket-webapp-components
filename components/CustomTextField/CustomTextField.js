/* eslint-disable react/prop-types */
import React, { useState, useEffect, useCallback } from "react";
import PropTypes from "prop-types";
import NumberFormat from "react-number-format";
import { makeStyles, TextField } from "@material-ui/core";
import textFieldStyle from "./textFieldStyle";
import { useDebounce } from "use-debounce";
import { curry } from "ramda";
import { emptyFunction, emptyString } from "../../utils/constants";

const useStyles = makeStyles(textFieldStyle);

const isAllowed = values =>
  !(values.floatValue && values.floatValue.toString().includes("e"));

function NumberFormatCustom(props) {
  const {
    inputRef,
    onChange,
    decimalScale,
    fixedDecimalScale,
    thousandSeparator,
    language,
    ...other
  } = props;
  let formatter = new Intl.NumberFormat(language);

  var thousandSep = thousandSeparator
    ? formatter.format(1111).replace(/1/g, "")
    : thousandSeparator;
  var decimalSeparator = formatter.format(1.1).replace(/1/g, "");
  const valueIsNumericString =
    !!props.value &&
    typeof props.value === "string" &&
    !Number.isNaN(Number(props.value));

  const handleValueChange = useCallback(values => onChange(values.floatValue), [
    onChange
  ]);

  return (
    <NumberFormat
      {...other}
      getInputRef={inputRef}
      style={{ textAlign: "right" }}
      onValueChange={handleValueChange}
      isAllowed={isAllowed}
      decimalScale={decimalScale}
      fixedDecimalScale={fixedDecimalScale}
      thousandSeparator={thousandSep}
      decimalSeparator={decimalSeparator}
      isNumericString={valueIsNumericString}
      //format = {customFormat}
    />
  );
}

NumberFormatCustom.propTypes = {
  inputRef: PropTypes.func.isRequired,
  onChange: PropTypes.func.isRequired,
  decimalScale: PropTypes.number.isRequired,
  fixedDecimalScale: PropTypes.bool,
  thousandSeparator: PropTypes.oneOfType([PropTypes.string, PropTypes.bool])
};

NumberFormatCustom.defaultProps = {
  decimalScale: 2,
  fixedDecimalScale: true,
  thousandSeparator: true
};

// TODO: this is for backwards compatibility (onTextBoxChange) ... remove onTextBoxChange and keep only value
const setValueToDesiredFormat = curry((isNumeric, value) =>
  isNumeric ? value : { target: { value } }
);
const getValueFromDesiredFormat = curry((isNumeric, e) =>
  isNumeric ? e : e?.target.value
);

function CustomTextField({
  isNumeric,
  customInputProps,
  inputProps,
  endAdornment,
  startAdornment,
  InputLabelProps,
  className,
  value,
  onChange,
  debounceBy,
  ...rest
}) {
  const classes = useStyles();

  // props applied to the Input element
  const customMuiInput = isNumeric
    ? {
        inputComponent: NumberFormatCustom,
        endAdornment,
        startAdornment,
        ...customInputProps
      }
    : {
        endAdornment,
        startAdornment,
        ...customInputProps
      };

  // attributes applied to the input element
  const customReactInput = {
    ...inputProps,
    className: `${classes.input} ${
      inputProps?.className ? inputProps.className : emptyString
    }`
  };

  const [localValue, setLocalValue] = useState(value);
  const [debouncedValue] = useDebounce(localValue, debounceBy);

  // eslint-disable-next-line react-hooks/exhaustive-deps
  useEffect(() => {
    debouncedValue !== value &&
      onChange(setValueToDesiredFormat(isNumeric)(debouncedValue));
  }, [debouncedValue]);
  // eslint-disable-next-line react-hooks/exhaustive-deps
  useEffect(() => {
    value !== localValue && setLocalValue(value);
  }, [value]);

  const handleChange = useCallback(
    e => setLocalValue(getValueFromDesiredFormat(isNumeric)(e)),
    [isNumeric]
  );

  return (
    <TextField
      onChange={handleChange}
      value={localValue}
      {...rest}
      className={classes.textField + " " + className}
      InputProps={customMuiInput}
      inputProps={customReactInput}
      InputLabelProps={{
        className: classes.label,
        ...InputLabelProps
      }}
    />
  );
}

CustomTextField.defaultProps = {
  isNumeric: false,
  debounceBy: 0,
  onChange: emptyFunction
};

CustomTextField.propTypes = {
  /**
   * Override or extend the styles applied to the component.
   */
  className: PropTypes.string,
  /**
   * If true, the input will accept only numeric values.
   */
  isNumeric: PropTypes.bool,
  /**
   * Other properties you can provide to the Input component.
   */
  customInputProps: PropTypes.object,
  /**
   * Attributes applied to the input element.
   */
  inputProps: PropTypes.object,
  /**
   * End adornment of componenent. (Usually an InputAdornment from material-ui)
   */
  endAdornment: PropTypes.any,
  /**
   * Start adornment of componenent. (Usually an InputAdornment from material-ui)
   */
  startAdornment: PropTypes.any,
  /**
   * Props applied to the InputLabel element.
   */
  InputLabelProps: PropTypes.object,
  /**
   * The value of the `input` element, required for a controlled component.
   */
  value: PropTypes.any,
  /**
   * Callback fired when the value is changed.
   *
   * @param {object} event The event source of the callback.
   * You can pull out the new value by accessing `event.target.value` (string).
   */
  onChange: PropTypes.func,
  /**
   * The delay of debouncing.
   */
  debounceBy: PropTypes.number,
  /**
   * The current language, preferably taken from the i18next (i18.language) or another internationalization library
   */
  language: PropTypes.string
};

export default CustomTextField;
