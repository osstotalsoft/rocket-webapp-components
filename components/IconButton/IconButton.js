import React from "react";
import PropTypes from "prop-types";
import cx from "classnames";
import { IconButton, makeStyles, deprecatedPropType } from "@material-ui/core";
import Tooltip from '../Tooltip/Tooltip';
import iconButtonStyle from "./iconButtonStyle";
import { dissoc } from 'ramda'

const useStyles = makeStyles(iconButtonStyle);

// eslint-disable-next-line react/prop-types
const Button = ({ children, color, className, ...rest }) => {
  const classes = useStyles();
  const buttonClasses =
    classes.button +
    cx({
      [" " + classes[color]]: color,
      [" " + className]: className
    });
  const restData = dissoc('title', { ...rest })
  return <IconButton {...restData} className={buttonClasses}>
    {children}
  </IconButton>
}

function CustomIconButton({ tooltip, ...rest }) {
  const title = tooltip ? tooltip : rest?.title
  return title ? (
    <Tooltip title={title}>
      <Button {...rest} />
    </Tooltip>
  ) : (
    <Button {...rest} />
  )
}

CustomIconButton.defaultProps = {
  color: "theme",
  size: "medium"
}

CustomIconButton.propTypes = {
  /**
   * The color of the button
   */
  color: PropTypes.oneOf([
    "primary",
    "info",
    "theme",
    "themeNoBackground",
    "themeWithBackground",
    "success",
    "warning",
    "danger",
    "rose",
    "white",
    "defaultNoBackground",
    "primaryNoBackground",
    "infoNoBackground",
    "successNoBackground",
    "warningNoBackground",
    "dangerNoBackground",
    "roseNoBackground",
    "themeWithGradient"
  ]),
  /**
  * The size of the button
  */
  size: PropTypes.oneOf([
    "small",
    "medium"
  ]),
  /**
  * Override or extend the styles applied to the component
  */
  customClass: deprecatedPropType(
    PropTypes.string,
    'Use `className` instead.',
  ),
  /**
  * Override or extend the styles applied to the component
  */
  className: PropTypes.string,
  /**
  * If true, button will be disabled. Default is set to false
  */
  disabled: PropTypes.bool,
  /**
  * The text to be displayed when the user hover over the button
  */
  tooltip: PropTypes.string,
  /**
   * The content of the button
   */
  children: PropTypes.node
};

export default CustomIconButton;