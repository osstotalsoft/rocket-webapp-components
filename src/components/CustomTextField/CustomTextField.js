/* eslint-disable react/prop-types */
import React, { useState, useEffect, useCallback } from "react";
import PropTypes from "prop-types";
import NumberFormat from "react-number-format";
import { makeStyles, TextField, deprecatedPropType } from "@material-ui/core";
import textFieldStyle from "./textFieldStyle";
import { useDebounce } from "use-debounce";
import { curry, is, head } from "ramda";
import { emptyFunction, emptyString } from "../../utils/constants";

const useStyles = makeStyles(textFieldStyle);

const isAllowed = values =>
  !(values.floatValue && values.floatValue.toString().includes("e"));

function NumberFormatCustom(props) {
  const {
    inputRef,
    onChange,
    decimalScale,
    fixedDecimalScale,
    thousandSeparator,
    language,
    ...other
  } = props;
  let formatter = new Intl.NumberFormat(language);

  var thousandSep = thousandSeparator
    ? formatter.format(1111).replace(/1/g, "")
    : thousandSeparator;
  var decimalSeparator = formatter.format(1.1).replace(/1/g, "");
  const valueIsNumericString =
    !!props.value &&
    typeof props.value === "string" &&
    !Number.isNaN(Number(props.value));

  const handleValueChange = useCallback(values => onChange(values.floatValue), [
    onChange
  ]);

  return (
    <NumberFormat
      {...other}
      getInputRef={inputRef}
      style={{ textAlign: "right" }}
      onValueChange={handleValueChange}
      isAllowed={isAllowed}
      decimalScale={decimalScale}
      fixedDecimalScale={fixedDecimalScale}
      thousandSeparator={thousandSep}
      decimalSeparator={decimalSeparator}
      isNumericString={valueIsNumericString}
      //format = {customFormat}
    />
  );
}

NumberFormatCustom.propTypes = {
  inputRef: PropTypes.func.isRequired,
  onChange: PropTypes.func.isRequired,
  decimalScale: PropTypes.number.isRequired,
  fixedDecimalScale: PropTypes.bool,
  thousandSeparator: PropTypes.oneOfType([PropTypes.string, PropTypes.bool])
};

NumberFormatCustom.defaultProps = {
  decimalScale: 2,
  fixedDecimalScale: true,
  thousandSeparator: true
};

// TODO: this is for backwards compatibility (onTextBoxChange) ... remove onTextBoxChange and keep only value
const setValueToDesiredFormat = curry((isNumeric, value) =>
  isNumeric ? value : { target: { value } }
);
const getValueFromDesiredFormat = curry((isNumeric, e) =>
  isNumeric ? e : e?.target.value
);

function CustomTextField({
  isNumeric,
  customInputProps,
  inputProps,
  InputProps,
  endAdornment,
  startAdornment,
  InputLabelProps,
  className,
  value,
  onChange,
  debounceBy,
  helperText,
  ...rest
}) {
  const classes = useStyles();

  // props applied to the Input element
  const customMuiInput = isNumeric
    ? {
        inputComponent: NumberFormatCustom,
        endAdornment,
        startAdornment
      }
    : {
        endAdornment,
        startAdornment,
        ...InputProps
      };

  // attributes applied to the input element
  const customReactInput = {
    ...inputProps,
    className: `${classes.input} ${
      inputProps?.className ? inputProps.className : emptyString
    }`
  };

  const [localValue, setLocalValue] = useState(value);
  const [debouncedValue] = useDebounce(localValue, debounceBy);

  useEffect(() => {
    debouncedValue !== value &&
      onChange(setValueToDesiredFormat(isNumeric)(debouncedValue));
  }, [debouncedValue]); // eslint-disable-line react-hooks/exhaustive-deps

  useEffect(() => {
    value !== localValue && setLocalValue(value);
  }, [value]); // eslint-disable-line react-hooks/exhaustive-deps

  const handleChange = useCallback(
    e => setLocalValue(getValueFromDesiredFormat(isNumeric)(e)),
    [isNumeric]
  );

  return (
    <TextField
      onChange={handleChange}
      value={localValue}
      helperText={is(Array, helperText) ? head(helperText) : helperText}
      {...rest}
      className={className ? `${classes.textField} ${className}` : classes.textField}
      InputProps={customMuiInput}
      inputProps={customReactInput}
      InputLabelProps={{
        className: classes.label,
        ...InputLabelProps
      }}
    />
  );
}

CustomTextField.defaultProps = {
  isNumeric: false,
  debounceBy: 0,
  onChange: emptyFunction
};

CustomTextField.propTypes = {
  /**
   * Override or extend the styles applied to the component.
   */
  className: PropTypes.string,
  /**
   * If true, the input will accept only numeric values.
   */
  isNumeric: PropTypes.bool,
  /**
   * Other properties you can provide to the component.
   */
  customInputProps: deprecatedPropType(PropTypes.object, "Use `inputProps` instead."),
  /**
   * Attributes applied to the input element.
   * For the numeric input, you can provide properties like thousandSeparator, decimalScale and allowNegative.
   */
  inputProps: PropTypes.object,
  /**
   * Other properties you can provide to the Input component.
   */
  InputProps: PropTypes.object,
  /**
   * End adornment of componenent. (Usually an InputAdornment from material-ui)
   */
  endAdornment: PropTypes.any,
  /**
   * Start adornment of componenent. (Usually an InputAdornment from material-ui)
   */
  startAdornment: PropTypes.any,
  /**
   * Props applied to the InputLabel element.
   */
  InputLabelProps: PropTypes.object,
  /**
   * The value of the `input` element, required for a controlled component.
   */
  value: PropTypes.any,
  /**
   * The content of the helper under the input.
   */
  helperText: PropTypes.node,
  /**
   * Callback fired when the value is changed.
   *
   * @param {object} event The event source of the callback.
   * You can pull out the new value by accessing `event.target.value` (string).
   */
  onChange: PropTypes.func,
  /**
   * The delay of debouncing.
   */
  debounceBy: PropTypes.number,
  /**
   * The current language, preferably taken from the i18next (i18.language) or another internationalization library
   */
  language: PropTypes.string
};

export default CustomTextField;
