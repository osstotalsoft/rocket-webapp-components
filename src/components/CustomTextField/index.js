import CustomTextField from "./CustomTextField";
export default CustomTextField;