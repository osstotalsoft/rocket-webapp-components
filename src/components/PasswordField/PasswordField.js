import React, { useState, useCallback } from "react";
import CustomTextField from "../CustomTextField";
import { InputAdornment } from "@material-ui/core";
import Tooltip from '../Tooltip/Tooltip';
import { Visibility, VisibilityOff } from "@material-ui/icons";
import IconButton from "../IconButton";
import PropTypes from "prop-types";

const PasswordField = ({ showPasswordText, hidePasswordText, ...rest }) => {
  const [showPassword, setShowPassword] = useState(false);
  const tooltip = showPassword ? hidePasswordText : showPasswordText;

  const handleClickShowPassword = useCallback(_ => {
    setShowPassword(x => !x);
  }, []);

  const handleMouseDownPassword = useCallback(event => {
    event.preventDefault();
  }, []);

  return (
    <CustomTextField
      {...rest}
      inputProps={{
        type: showPassword ? "text" : "password"
      }}
      endAdornment={
        <InputAdornment position="end">
          <IconButton
            aria-label="toggle password visibility"
            onClick={handleClickShowPassword}
            onMouseDown={handleMouseDownPassword}
          >
            <Tooltip title={tooltip}>
              {showPassword ? <Visibility /> : <VisibilityOff />}
            </Tooltip>
          </IconButton>
        </InputAdornment>
      }
    />
  );
};

PasswordField.defaultProps = {
  hidePasswordText: "Hide password",
  showPasswordText: "Show password text"
};

PasswordField.propTypes = {
/**
* The text to be displayed on tooltip when the user wants to hide the password
*/
  hidePasswordText: PropTypes.string,
/**
* The text to be displayed on tooltip when the user wants to show the password
*/
  showPasswordText: PropTypes.string
};

export default PasswordField;
