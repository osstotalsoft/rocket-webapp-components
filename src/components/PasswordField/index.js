import PasswordField from "./PasswordField";
export default PasswordField;