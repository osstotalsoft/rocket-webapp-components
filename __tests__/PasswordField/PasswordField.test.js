import React from "react";
import renderer from "react-test-renderer";
import PasswordField from "../../components/PasswordField/PasswordField";
import { mount } from 'enzyme';
import IconButton from "../../components/IconButton/IconButton";
import CustomTextField from "../../components/CustomTextField/CustomTextField";

describe("PasswordField", () => {
  it("Snapshot", () => {
    const component = renderer.create(
      <PasswordField
        hidePasswordText="Hide password"
        showPasswordText="Show password text"
      />
    );

    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });

  it("default props should be provided", () => {
    const wrapper = mount(
      <PasswordField />
    );

    expect(wrapper.props().hidePasswordText).toBe("Hide password")
    expect(wrapper.props().showPasswordText).toBe("Show password text")
  });

  it("after pressing the button the input should turn into a text field", () => {
    const wrapper = mount(
      <PasswordField
        hidePasswordText="Hide password"
        showPasswordText="Show password text" />
    );

    expect(wrapper.find(CustomTextField).props().inputProps.type).toBe('password')

    wrapper.find(IconButton).simulate('click');

    expect(wrapper.find(CustomTextField).props().inputProps.type).toBe('text')
  });
});
