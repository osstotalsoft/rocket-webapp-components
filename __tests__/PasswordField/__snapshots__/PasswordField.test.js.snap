// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PasswordField Snapshot 1`] = `
<div
  className="MuiFormControl-root MuiTextField-root"
>
  <div
    className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl MuiInputBase-adornedEnd"
    onClick={[Function]}
  >
    <input
      aria-invalid={false}
      autoFocus={false}
      className="MuiInputBase-input MuiInput-input makeStyles-input-2  MuiInputBase-inputAdornedEnd"
      disabled={false}
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      required={false}
      type="password"
    />
    <div
      className="MuiInputAdornment-root MuiInputAdornment-positionEnd"
    >
      <button
        aria-label="toggle password visibility"
        className="MuiButtonBase-root MuiIconButton-root makeStyles-button-3 makeStyles-theme-18"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiIconButton-label"
        >
          <span
            aria-describedby={null}
            className=""
            onBlur={[Function]}
            onFocus={[Function]}
            onMouseLeave={[Function]}
            onMouseOver={[Function]}
            onTouchEnd={[Function]}
            onTouchStart={[Function]}
            title="Show password text"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 7c2.76 0 5 2.24 5 5 0 .65-.13 1.26-.36 1.83l2.92 2.92c1.51-1.26 2.7-2.89 3.43-4.75-1.73-4.39-6-7.5-11-7.5-1.4 0-2.74.25-3.98.7l2.16 2.16C10.74 7.13 11.35 7 12 7zM2 4.27l2.28 2.28.46.46C3.08 8.3 1.78 10.02 1 12c1.73 4.39 6 7.5 11 7.5 1.55 0 3.03-.3 4.38-.84l.42.42L19.73 22 21 20.73 3.27 3 2 4.27zM7.53 9.8l1.55 1.55c-.05.21-.08.43-.08.65 0 1.66 1.34 3 3 3 .22 0 .44-.03.65-.08l1.55 1.55c-.67.33-1.41.53-2.2.53-2.76 0-5-2.24-5-5 0-.79.2-1.53.53-2.2zm4.31-.78l3.15 3.15.02-.16c0-1.66-1.34-3-3-3l-.17.01z"
              />
            </svg>
          </span>
        </span>
        <span
          className="MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</div>
`;
